copyspecial.py

```diff
@@ -16,29 +16,60 @@
 import argparse
 
 # This is to help coaches and graders identify student assignments
-__author__ = "???"
+__author__ = "knmarvel with madarp"
 
 
-# +++your code here+++
-# Write functions and modify main() to call them
+def get_special_paths(dir):
+    """takes a directory and returns a list of the absolute
+    paths of the special files in the given directory"""
+    spec_paths = []
+    for root, dirs, files in os.walk(dir):
+        for name in files:
+            if re.search("__(.+?)__", name):
+                if os.path.join(os.path.abspath(dir), name) not in spec_paths:
+                    spec_paths.append(os.path.join(os.path.abspath(dir), name))
+                else:
+                    return "ERROR"
```

Comment:
nice check.  it would be helpful on larger programs to return descriptive text a little longer than just "ERROR".  maybe something like "Error while trying to make list of special programs: duplicate filename found."

Reply:
I see below where you put the error message in the main() function.  It makes more sense to me to have the error message be with the function where the error is occurring.